phoenix-core-client/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -297,7 +297,9 @@ public static ColumnResolver getIndexResolver(SingleTableStatement statement,
 
     public static ColumnResolver getResolver(SingleTableStatement statement, PhoenixConnection connection, Map<String, UDFParseNode> udfParseNodes)
             throws SQLException {
-        SingleTableColumnResolver visitor = new SingleTableColumnResolver(connection, statement.getTable(), true, 0, udfParseNodes);
+        SingleTableColumnResolver visitor
```

Comment:
I see that we are calling this resolver from `MetaDataClient#createIndex `and `CreateIndexCompiler#compile` method. So we want to bypass cache for createIndex and resolve table always. So lets rename this method to `getResolverForCreateIndex` and add a comment on why we are setting `alwaysHitServer` to true.